### PR TITLE
CLOUDP-115398: [Atlas CLI] Improve CLI login experience

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-115398

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

In this PR we're including the following improvements to the login experience:

- Showing the message "you have already set the programmatic keys for this profile, run `atlas auth login --profile <profile_name>` to use your username and password on a new profile" when the user is trying to use auth login on a profile which has programmatic keys
- Enabling the flag `--loginOnly   Skip profile configuration.`

```
./bin/atlas auth login --help
Authenticate with MongoDB Atlas.

Usage:
  atlas auth login [flags]

Examples:
  To start the interactive setup:
  $ mongocli auth login


Flags:
      --gov         Log in to Atlas for Government.
  -h, --help        help for login
      --loginOnly   Skip profile configuration.
      --noBrowser   Don't try to open a browser session.

Global Flags:
  -P, --profile string   Profile to use from your configuration file.
```

```
/bin/atlas auth login --loginOnly
Error: you have already set the programmatic keys for this profile.

Run 'atlas auth login --profile <profile_name>' to use your username and password on a new profile
```

### CODE REVIEW

File: internal/cli/auth/login.go
Comment (gssbzn) [inline internal/cli/auth/login.go:92]: what will happen when the token expires?

Comment (andreaangiolillo) [reply internal/cli/auth/login.go:92]: users will have to first run `auth logout` and then run `auth login` as I think they are doing right now

**Review by gssbzn — CHANGES_REQUESTED** [review]
checking the token has a subject is not a good check, tokens expire which means users need to re-authenticate, you can check if the token has expired, we have logic for that but then we have a refresh token which sadly lacks expiration embedded and you can only verify it expired by trying to use it 
TL;DR preventing users to authenticate based on there being a token is no bueno and may lock out users with expired tokens

File: internal/cli/auth/login.go
Comment (andreaangiolillo) [reply internal/cli/auth/login.go:92]: I see how it makes sense to check if the token is valid

Comment (gssbzn) [reply internal/cli/auth/login.go:92]: right now I don't have to login, I just re-login

Comment (gssbzn) [reply internal/cli/auth/login.go:92]: my recommendation to discuss with @dianchenghu and @boooczek is to ask users if they want to reauthenticate,
```bash
You are already authenticated as <>: do you want to re authenticate(Y/n): _
```

Comment (boooczek) [reply internal/cli/auth/login.go:92]: I understand what we're discussing here is case when token expieres. Then when user types in "atlas login" the flow is as usual, i.e. they go through auth process in the browser.
@gssbzn why users would want to re-authenticate?

Comment (gssbzn) [reply internal/cli/auth/login.go:92]: > why users would want to re-authenticate? Is it because their token might expire soon?
- I may have multiple atlas account (I do, personal and professional) and I may want to switch between them
- See my general comment on token expiration:  tokens expire which means users need to re-authenticate, you can check if the token has expired, we have logic for that but then we have a refresh token which sadly lacks expiration embedded and you can only verify it expired by trying to use it, there are two tokens one we know when it expires the second one we don't until we try to use it, we can't change how that second token works

File: internal/cli/auth/login.go
Comment (andreaangiolillo) [reply internal/cli/auth/login.go:209]: Leaving this comment open so that doc team can decide how to call this flag

File: internal/cli/root/atlas/builder.go
Comment (andreaangiolillo) [reply internal/cli/root/atlas/builder.go]: makes sense. Updated

**Comment by andreaangiolillo** [comment]
evergreen retry

File: internal/cli/root/atlas/builder.go
Comment (tibulca) [inline internal/cli/root/atlas/builder.go:104]: not a strong opinion here (since I've seen both options in the past), but if refreshAccessToken fails do we want to fail the flow? Maybe the existing credentials are still valid. And if the credentials are not valid anymore the flow will be interrupted at by line `return validate.Credentials()`

Comment (tibulca) [reply internal/cli/root/atlas/builder.go:104]: ignore this, checked the RefreshAccessMethod and if the token is valid it doesn't try to refresh it (I thought it will try to refresh it if x% of it's validity passed)
```
	if current.Valid() {
		return nil
	}
```

**Review by tibulca — APPROVED** [review]

File: internal/cli/auth/login.go
Comment (sarahsimpers) [reply internal/cli/auth/login.go:209]: I do think skipConfig is a better name for it.

**Review by sarahsimpers — DISMISSED** [review]
LGTM % comment on flag name, once this gets merged I will change the Atlas CLI docs to reflect this as the quickest getting started method. Thanks!

**Review by johansteyn — APPROVED** [review]
LGTM

**Review by tibulca — APPROVED** [review]

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**
